relay-general/src/store/transactions/processor.rs

```diff
@@ -362,10 +362,8 @@ impl Processor for TransactionsProcessor<'_> {
                 event.transaction_info.value_mut(),
             )?;
 
-            if !matches!(
-                event.get_transaction_source(),
-                &TransactionSource::Sanitized
-            ) && self.name_config.mark_scrubbed_as_sanitized
+            if matches!(event.get_transaction_source(), &TransactionSource::Url)
+                && self.name_config.mark_scrubbed_as_sanitized
```

Comment:
As discussed offline, we might want to add an an additional condition which is `scrub_identifiers_changed_something OR rename_rules.len() > 0`, to prevent projects that do not have any rules yet from marking everything as sanitized. But that is actually out of scope for this PR.